code/modules/organs/internal/liver/liver.dm

```diff
@@ -60,4 +60,26 @@
 	var/reagent_efficiency = 1
 	if(reagent_id in reagent_efficiencies)
 		reagent_efficiency = reagent_efficiencies[reagent_id]
-	H.reagents.remove_reagent(reagent_id, metabolism * efficiency * reagent_efficiency)
\ No newline at end of file
+	H.reagents.remove_reagent(reagent_id, metabolism * efficiency * reagent_efficiency)
+
+/datum/organ/internal/liver/diona
+	name = "filtration xylem"
+
+/datum/organ/internal/liver/diona/metabolize_reagent(var/reagent_id, var/metabolism)
+	..()
+	var/mob/living/carbon/human/H = owner
+	switch(reagent_id)
+		if(HYPERZINE || NUKA_COLA) // these are highly addictive and hallucinogenic drugs for diona - think sensory overload mixed with skooma
```

Comment:
This is the wrong place to put species-specific drug effects. You can either check the species in the `on_mob_life()` code (blech) or check that the second parameter, `alien`, is `IS_DIONA`, and then put the behaviour under that.